### DISCUSSION TITLE
No more IB

### DISCUSSION THREAD

**Discussion by mfrener** [stoqey/ib#137] [General]
Unfortunately I have to announce that can no longer work on this project.
I will still help out if new developers want to join and have questions, but I wont be able to use the lib on my own, so no more point in spending time on it.

What happend?
I have been using IB platfrom via an Introducing Broker (https://www.captrader.com/).
With the new year I wanted to create an addional account to further partion my portfolios, but I would go with IBKR directly this time.
So I created a new account request at IBKR directly, which causes an approval request at my Introducing Broker, which caused an email to me, starting with `Aufgrund von Vereinbarungen zwischen Interactive Brokers und seinem Introducing Broker CapTrader ist ein Wechsel von CapTrader zu Interactive Brokers nur nach der Kontoschließung bei CapTrader und einer Wartefrist von 6 Monaten bis Neueröffnung möglich.` Means: CapTrader won't allow me to open an account with IBKR directly. And it's not only that, if I decide to close my account with them, they are going to block me for an addtional 6 more months.
What would you do now?
Stay in a business relationship, with a company that will kick you in the back after you out the door, by blocking you on 3rd parties? Or leave asap?
I opted for the second, have closed down my investment account already and are about to migate to a new platform for the trading account, preparing to close down with IBKR completly.
I cannot say yet if I will give them a new try after the 6 month lock-out period or not, will probably depend on if I like the new platfrom.

What can you learn from that?
- Be very carefull with IB Introducing Brokers. If you already signed up with one, maybe create an account with IB directly, just to see if you sit on the same "if you leave, I ban you" lock.
- Stay far way from https://www.captrader.com
- Have a backup plan ready. While this brings some huzzle now, it's no real problem. I have a new platform ready already as it was my "fallback" anyhow. But it reminds once more, if you want to survie on this bizz, never go all-in, not even with your broker.

So what to say more.. hope this projects find new devs and will stay alive.
Wish you all the best.

**Comment by electic** [comment]
What broker did you go with?

**Reply by mfrener** [reply]
- Investement / buy&hold is Degiro.
- Option & Futures trading https://www.ibroker.it/ (a spanish/italian Futures broker, has no API but at webapp that uses one.. so kind of :D)
- Equity trading I just started with (was the reason I wanted to requested 3rd account and caused me to noticie that my broker is a sucker) - have not decided yet, still trying out, but right now looks like it will be https://alpaca.markets/

**Comment by Josh505050** [comment]
I'm using captrader, too and found your ib-gateway-docker, Thank you for that.
However I need some help to use it in combination with Webhooks from Tradingview. 
I wonder if you can help me with that?

Best Regards 
Josh

**Reply by mfrener** [reply]
Never did anything with webhooks, so not sure if can help on that (in theory it should only be about to bridge the correct ports so your http server is reachable from outside). 
My ib docker is no more in use nor updated/maintained, but have a look on that https://github.com/extrange/ibkr-docker if you search for alternatives

**Reply by Josh505050** [reply]
Ok thank you, I will have a look on it

**Reply by loixlab** [reply]
Thanks for sharing your experience.
FYI there is also this project https://github.com/Voyz/ibeam that exposes a Docker image for the [Interactive Brokers Client Portal Web API Gateway.](https://interactivebrokers.github.io/cpwebapi/)
